### PR TITLE
Add Application Gateway Re-encryption Template

### PR DESCRIPTION
**Pull request by suneetnangia**
### Contributing guide

https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md
### Changelog

*
*
*
*
*

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@suneetnangia__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
Hi __@suneetnangia__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sunangi). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
Hi __@suneetnangia__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sunangi). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+              ]
+            }
+          }
+        ],
+        "httpListeners": [
+          {
+            "name": "appGatewayHttpListener",
+            "properties": {
+              "FrontendIPConfiguration": {
+                "Id": "[concat(variables('applicationGatewayID'), '/frontendIPConfigurations/appGatewayFrontendIP')]"
```
Comment (bmoore-msft) [inline 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json]: use resourceId()

File: 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+              },
+              "FrontendPort": {
+                "Id": "[concat(variables('applicationGatewayID'), '/frontendPorts/appGatewayFrontendPort')]"
```
Comment (bmoore-msft) [inline 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json]: use resourceId()

File: 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+              },
+              "Protocol": "Https",
+              "SslCertificate": {
+                "Id": "[concat(variables('applicationGatewayID'), '/sslCertificates/appGatewaySslCert')]"
```
Comment (bmoore-msft) [inline 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json]: use resourceId()

File: 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+            }
+          }
+        ],
+        "requestRoutingRules": [
+          {
+            "Name": "rule1",
+            "properties": {
+              "RuleType": "Basic",
+              "httpListener": {
+                "id": "[concat(variables('applicationGatewayID'), '/httpListeners/appGatewayHttpListener')]"
```
Comment (bmoore-msft) [inline 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json]: use resourceId()

File: 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+              },
+              "backendAddressPool": {
+                "id": "[concat(variables('applicationGatewayID'), '/backendAddressPools/appGatewayBackendPool')]"
```
Comment (bmoore-msft) [inline 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json]: use resourceId()

File: 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+              },
+              "backendHttpSettings": {
+                "id": "[concat(variables('applicationGatewayID'), '/backendHttpSettingsCollection/appGatewayBackendHttpSettings')]"
```
Comment (bmoore-msft) [inline 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json]: use resourceId()

File: 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+        "Standard_Small",
+        "Standard_Medium",
+        "Standard_Large"
+      ],
+      "defaultValue": "Standard_Medium",
+      "metadata": {
+        "description": "Sku Name"
+      }
+    },
+    "capacity": {
```
Comment (bmoore-msft) [inline 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json]: move to variable

File: 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+          }
+        ],
+        "authenticationCertificates": [
+          {
+            "name": "whitelistCert",
+            "properties": {
+              "data": "[parameters('backendCertData')]",
+              "backendHttpSettings": [
+                {
+                  "id": "[concat(variables('applicationGatewayID'), '/backendHttpSettingsCollection/appGatewayBackendHttpSettings')]"
```
Comment (bmoore-msft) [inline 101-application-gateway-public-ip-ssl-reencrypt/azuredeploy.json]: use resourceId()

**Comment by bmoore-msft** [comment]
closing due to no activity

**Pull request closed without merging**